sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+                    };
+                };
+                readonly Websites: {
+                    readonly type: "array";
+                    readonly items: {
+                        readonly type: "string";
+                        readonly description: "Website";
+                        readonly example: "https://www.contoso.com";
+                    };
+                };
+            };
+        };
+    };
+};
+
+// @public
+export interface CommonClientOptions extends PipelineOptions {
+    apiVersion?: FormRecognizerApiVersion;
+    stringIndexType?: StringIndexType;
+}
+
+// @public
+export type ContentType = "application/octet-stream" | "application/pdf" | "image/bmp" | "image/jpeg" | "image/png" | "image/tiff";
+
+// @public
+export interface CopyAuthorization {
+    accessToken: string;
+    expirationDateTime: Date;
+    targetModelId: string;
+    targetModelLocation: string;
+    targetResourceId: string;
+    targetResourceRegion: string;
+}
+
+// @public
+export interface CopyModelOptions extends OperationOptions, PollerOptions<TrainingPollOperationState> {
+}
+
+// @public
+export interface CustomDocumentModelsInfo {
+    count: number;
+    limit: number;
+}
+
+// @public
+export interface DateFieldSchema {
+    // (undocumented)
+    readonly type: "date";
+}
+
+// @public
+export interface DeleteModelOptions extends OperationOptions {
+}
+
+// @public
+export interface DocTypeInfo {
+    description?: string;
+    fieldConfidence?: {
+        [propertyName: string]: number;
+    };
+    fieldSchema: {
+        [propertyName: string]: DocumentFieldSchema;
+    };
+}
+
+// @public
+export class DocumentAnalysisClient {
+    constructor(endpoint: string, credential: TokenCredential, options?: DocumentAnalysisClientOptions);
+    constructor(endpoint: string, credential: KeyCredential, options?: DocumentAnalysisClientOptions);
+    // @internal
+    constructor(endpoint: string, credential: KeyCredential | TokenCredential, options?: DocumentAnalysisClientOptions);
+    beginAnalyzeDocuments(modelId: string, input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions): Promise<AnalysisPoller>;
+    beginAnalyzeDocuments<Document>(model: DocumentModel<Document>, input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<AnalyzeResult<Document>>): Promise<AnalysisPoller<AnalyzeResult<Document>>>;
+    beginExtractGenericDocument(input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<GenericDocumentResult>): Promise<AnalysisPoller<GenericDocumentResult>>;
+    beginExtractLayout(input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<LayoutResult>): Promise<AnalysisPoller<LayoutResult>>;
+}
+
+// @public
+export interface DocumentAnalysisClientOptions extends CommonClientOptions {
+}
+
+// @public
+export interface DocumentAnalysisPollOperationState<Result = AnalyzeResult<AnalyzedDocument>> extends PollOperationState<Result> {
```

Comment:
is there any other info the service returns that could be part of the state, e.g. the date and time when the job was created?

Reply:
Think there might be a lastUpdatedDateTime. I'll double-check to make sure we have all the properties of the operation status in here. I originally wrote this as `extends AnalyzeResultOperation, PollOperationState<Result>`, but there was a problem with that (they have conflicting keys) so I added the fields manually and wrote the transform.